File: LeadToOffer/21_StackPushAndPop.cpp
class Solution {
public:
    bool IsPopOrder(vector<int> pushV,vector<int> popV) {
        if (pushV.size() != popV.size())
            return false;
        if (pushV.size()==0 && popV.size()==0)
            return false;

        stack<int> temp;
        int i = 0;
        int j = 0;
        while (i<pushV.size()) {
            temp.push(pushV[i]);
            i++;
            while (!temp.empty() && popV[j]==temp.top()) {
                temp.pop();
                j++;
            }
        }

        return temp.empty();
    }
};

File: LeadToOffer/21_StackPushAndPop.py
class Solution:
    def IsPopOrder(self, pushV, popV):
        if len(pushV)!=len(popV):
            return False
        if len(pushV)==0 and len(popV)==0:
            return False

        res = []
        i,j = 0,0
        while i<len(pushV):
            if len(res)!=0 and res[-1]==popV[j]:
                res.pop()
                j += 1
            else:
                res.append(pushV[i])
                i += 1
        while len(res)>0:
            if res[-1]!= popV[j]:
                return False
            res.pop()
            j += 1
        return True
